src/strategies/MangroveOffer.sol

```diff
@@ -260,13 +259,28 @@ abstract contract MangroveOffer is AccessControlled, IOfferLogic {
     return order.offer.gives() - order.wants;
   }
 
+  ///@notice converts Mangrove's error message string back into its original bytes32 type
+  ///@param reason the revert reason
+  function _repostStatus(string memory reason) internal pure returns (bytes32) {
```

Comment:
If I understand correctly then this method could be reduced to return bytes32(reason) except for the REPOST_FAILED case. Am I missing something?

Reply:
yes expect you can't cast a string to a bytes32...